src/Cli/dotnet/commands/dotnet-tool/install/ToolInstallGlobalOrToolPathCommand.cs

```diff
@@ -157,5 +171,39 @@ public override int Execute()
                     isUserError: false);
             }
         }
+
+        private async Task<string> ResolveAppHostSourceDirectoryAsync(string archOption, string targetFramework)
+        {
+            string rid;
+            var validRids = new string[] { "win-x64", "win-arm64", "osx-x64", "osx-arm64" };
+            if (string.IsNullOrEmpty(archOption))
+            {
+                if (!string.IsNullOrEmpty(targetFramework) && new NuGetFramework(targetFramework).Version < new Version("6.0")
+                    && (OperatingSystem.IsWindows() || OperatingSystem.IsMacOS()) && !RuntimeInformation.ProcessArchitecture.Equals(Architecture.X64))
+                {
+                    rid = OperatingSystem.IsWindows() ? "win-x64" : "osx-x64";
+                }
+                else
+                {
+                    // Use the default app host
+                    return null;
+                }
+            }
+            else
+            {
+                rid = CommonOptions.ResolveRidShorthandOptionsToRuntimeIdentifier(null, archOption);
+            }
+
+            if (!validRids.Contains(rid))
+            {
+                throw new GracefulException(string.Format(LocalizableStrings.InvalidRuntimeIdentifier, rid, string.Join(" ", validRids)));
+            }
+
+            var packageId = new PackageId($"microsoft.netcore.app.host.{rid}");
+            var packagePath = await _nugetPackageDownloader.DownloadPackageAsync(packageId);
```

Comment:
might want retry here

Reply:
also, we might want to pipe the 4,5 nuget options into here. If the user's nuget config has a bad feed, it would fail. And also we need to have apphost in one of the feed.